### ISSUE TITLE
surface plot not working

### ISSUE THREAD

**Issue by pfbuxton** [matlab2tikz/matlab2tikz#852]
Hello,

Here is a test code which fails:

<code>
clear;

x = linspace(0,10,100);
y = x.^2;
c = y.^5;

surface(...
  'XData',[x(:) x(:)],...
  'YData',[y(:) y(:)],...
  'ZData',zeros(length(x(:)),2),...
  'CData',[c(:) c(:)],...
  'FaceColor','none',...
  'EdgeColor','flat',...
  'Marker','none', ...
  'LineWidth',4);

matlab2tikz( 'test.tikz', 'standalone',true );

dos( [ 'pdflatex test.tikz"' ] );

</code>

**Comment by pfbuxton** [comment]
I have found an alternative which unfortunately also does not work (at least on my system):

```matlab
clear;

x = linspace(0,10,100);
y = x.^2;
c = y.^5;

patch( [x fliplr(x)],[y+1.0 fliplr(y)-1.0], [c fliplr(c)], 'linestyle','none' );

matlab2tikz( 'test.tikz', 'standalone',true );

dos( [ 'pdflatex test.tikz"' ] );
```

**Comment by miscco** [comment]
That one seems like a simple one.
isFaceColorFlat

is not defined in getPatchShape.

I will submit a patch for that

**Comment by miscco** [comment]
So the second one should be fine by defining isFaceColorFlat, however it seems the first one has some problems, as it does not compile correctly. It seems some color keys are incorrect:
```
! Package pgfkeys Error: The key '/tikz/color' requires a value. I am going to 
ignore this key.
```

The culprit has this form:
```latex
\addplot[%
mesh,
color,colormap={mymap}{[1pt] rgb(0pt)=(0.2081,0.1663,0.5292); rgb(1pt)=(0.211624,0.189781,0.577676); rgb(2pt)=(0.212252,0.213771,0.626971); 
...
rgb(62pt)=(0.9661,0.951443,0.0755333); rgb(63pt)=(0.9763,0.9831,0.0538)},mesh/rows=2]
table[row sep=crcr, point meta=\thisrow{c}] {%
```

**Comment by miscco** [comment]
Here the reswults for the first plot:
![issue_852_matlab](https://cloud.githubusercontent.com/assets/3017120/12624559/0410cf26-c530-11e5-8a6d-b723f2aedc82.jpg)

And the resulting pdf
![issue852_m2t](https://cloud.githubusercontent.com/assets/3017120/12624566/0a24a608-c530-11e5-8280-76b2214a3979.jpg)

**Comment by miscco** [comment]
@egeerardyn I have never looked at patch and msurface plots do you have any idea about that color?

**Comment by egeerardyn** [comment]
@okomarov has written most of the relevant code. I'm not an expert on the details either (and I don't often use such plots, to be honest)

**Comment by miscco** [comment]
So we end up in the 'flat' branch of
```
patchcolor2xcolor
``` 

However, the cdata we get is rather large (10e10) and we end in the following branch
```matlab
                    % Don't return anything meaningful and count on the caller
                    % to make something of it.
                    xcolor = [];
```

That seems like its intentional, so i would wait for @okomarov  to tell us whats up

**Comment by okomarov** [comment]
The short answer: I do not think we support CDATA in matrix form. 

The long answer:
All low-level color functions existed before I worked on `drawPatch()`, and they might need some re-organization, but the task is not an easy one.

With respect to patches (as a generic graphical object), there is considerable complexity. Also, there is not a 1to1 correspondence between the coloring schemes available in pgfplots and those implemented by Matlab patches. Often, the difference is not documented either. Hence, the current implementation of `drawPatch()` is in many ways hacky. 

@pfbuxton Are you trying to plot a line that changes color (and you found the solution in http://www.mathworks.com/matlabcentral/answers/5042-how-do-i-vary-color-along-a-2d-line)? If that is the case, the immediate workaround is to plot, in a loop, a line() for each segment.

**Comment by pfbuxton** [comment]
@okomarov yes thank you I was trying to do a line which changes colour. I have made a short function to do this:

```matlab
function [] = color_line_works_with_matlab2tikz( axes_handles, x, y, values )
    
    jet_values = jet(90);
    jet_x      = linspace( min(values), max(values), 90 );
    
    for i=1:length(x)-1
        plot( axes_handles, [x(i) x(i+1)], [y(i) y(i+1)], 'LineWidth',1.0, 'Color',interp1(jet_x,jet_values,values(i)) );
    end;
    
end
```

Thanks,
Peter
